src/histogram.ts

```diff
@@ -18,82 +18,113 @@ import { SummaryValue } from "@dynatrace/metric-utils";
-		return { count: 0, sum: 0, min: 0, max: 0 };
-	}
-
-	const { min, max } = getMinMax(point);
+	const { min, max, sum } = estimateOptionalProperties(point);
 
 	return {
 		count: point.value.count,
-		sum: point.value.sum,
+		sum,
 		max,
 		min
 	};
 }
 
-function getMinMax(point: DataPoint<Histogram>): { min: number; max: number } {
-	if (point.value.hasMinMax) {
-		return { min: point.value.min, max: point.value.max };
-	}
+function estimateSingleBucketHistogram(point: DataPoint<Histogram>): { min: number; max: number; sum: number } {
+	const sum = point.value.sum ?? 0;
+	// Only called after checking that count > 0
+	const mean = sum / point.value.count;
 
-	return estimateHistMinMax(point);
+	return { min: point.value.min ?? mean, max: point.value.max ?? mean, sum };
 }
 
-// Expect to be called only with points that contain actual data (point.value.count > 0)
-function estimateHistMinMax(point: DataPoint<Histogram>): { min: number; max: number } {
-	const { counts, boundaries } = point.value.buckets;
-	const avg = point.value.sum / point.value.count;
+function estimateOptionalProperties(point: DataPoint<Histogram>): { min: number; max: number; sum: number } {
+	// shortcut if min, max, and sum are provided
+	if (point.value.min != null && point.value.max != null && point.value.sum != null) {
+		return { min: point.value.min, max: point.value.max, sum: point.value.sum };
+	}
+
+	// Shortcut for 0 count histograms
+	if (point.value.count === 0) {
+		return { min: 0, max: 0, sum: 0 };
+	}
+
+	const counts = point.value.buckets.counts;
+	const boundaries = point.value.buckets.boundaries;
 
-	if (boundaries.length === 0 && counts[0] > 0) {
-		return { min: avg, max: avg };
+	// a single-bucket histogram is a special case
+	if (counts.length === 1) {
+		return estimateSingleBucketHistogram(point);
 	}
 
-	// Because we do not know the actual min and max, we estimate them based on the min and max non-empty bucket
-	let minCountIdx = -1;
-	let maxCountIdx = -1;
+	// If any of min, max, sum is not provided in the data point,
+	// loop through the buckets to estimate them.
+	// All three values are estimated in order to avoid looping multiple times
+	// or complicating the loop with branches. After the loop, estimates
+	// will be overridden with any values provided by the data point.
+	let foundNonEmptyBucket = false;
+	let min = 0;
+	let max = 0;
+	let sum = 0;
+
+	// Because we do not know the actual min, max, or sum, we estimate them based on non-empty buckets
 	for (let i = 0; i < counts.length; i++) {
-		if (counts[i] > 0) {
-			if (minCountIdx === -1) {
-				minCountIdx = i;
-			}
-			maxCountIdx = i;
+		// empty bucket.
+		if (counts[i] === 0) {
+			continue;
 		}
-	}
 
-	// no values found
-	if (minCountIdx === -1 || maxCountIdx === -1) {
-		return {
-			min: 0,
-			max: 0
-		};
-	}
+		// range for bucket counts[i] is bounds[i-1] to bounds[i]
 
-	let min: number;
-	let max: number;
+		// min estimation.
+		if (!foundNonEmptyBucket) {
```

Comment:
It's nice how close to the collector impl this now is :). Makes reviewing easier!